spark/core/main/scala/org/elasticsearch/spark/serialization/ScalaValueWriter.scala

```diff
@@ -62,10 +62,12 @@ class ScalaValueWriter(writeUnknownTypes: Boolean = false) extends JdkValueWrite
         generator.writeBeginObject()
         for ((k, v) <- m) {
           if (shouldKeep(parentField, k.toString)) {
-            generator.writeFieldName(k.toString)
-            val result = doWrite(v, generator, k.toString)
-            if (!result.isSuccesful) {
-              return result
+            if (v != None && v!= null && v != () | hasWriteNullValues) {
```

Comment:
This probably ought to use "||" rather than "|" right?

Reply:
I agree